PyStemmusScope/global_data/cci_landcover.py

```diff
@@ -0,0 +1,160 @@
+        time_range: Start and end time of the model run.
+        timestep: Desired timestep of the model, this is derived from the forcing data.
+            In a pandas-timedelta compatible format. For example: "1800S"
+
+    Returns:
+        Dictionary containing IGBP and LCCS land cover classes.
+    """
+    files_cci = list((global_data_dir / "landcover").glob("*.nc"))
+
+    if len(files_cci) == 0:
+        raise FileNotFoundError(
+            f"No netCDF files found in the folder '{global_data_dir / 'landcover'}'"
+        )
+
+    return extract_landcover_data(
+        files_cci=files_cci,
+        latlon=latlon,
+        time_range=time_range,
+        timestep=timestep,
+    )
+
+
+def extract_landcover_data(
+    files_cci: List[Path],
+    latlon: Union[Tuple[int, int], Tuple[float, float]],
+    time_range: Tuple[np.datetime64, np.datetime64],
+    timestep: str,
+) -> Dict[str, np.ndarray]:
+    """Extract the land cover data from the CCI netCDF files.
+
+    Args:
+        files_cci: List of CCI land cover files.
+        latlon: Latitude and longitude of the site.
+        time_range: Start and end time of the model run.
+        timestep: Desired timestep of the model, this is derived from the forcing data.
+            In a pandas-timedelta compatible format. For example: "1800S"
+
+    Returns:
+        Dictionary containing IGBP and LCCS land cover classes.
+    """
+    cci_dataset = xr.open_mfdataset(files_cci)
+
+    check_cci_dataset(cci_dataset, latlon, time_range)  # Assert spatial/temporal bounds
+
+    lat_bounds = cci_dataset["lat_bounds"].load()  # Load so that they are not
+    lon_bounds = cci_dataset["lon_bounds"].load()  #  dask arrays
+    lat_idx = np.logical_and(  # type: ignore
+        lat_bounds.isel(bounds=0) >= latlon[0], lat_bounds.isel(bounds=1) < latlon[0]
+    ).argmax(dim="lat")
+    lon_idx = np.logical_and(  # type: ignore
+        lon_bounds.isel(bounds=0) <= latlon[1], lon_bounds.isel(bounds=1) > latlon[1]
+    ).argmax(dim="lon")
+
+    lccs_id = cci_dataset.isel(lat=lat_idx, lon=lon_idx)["lccs_class"]
+
+    # If time is size 1, interp fails. Adding an extra datapoint prevents this.
+    if lccs_id["time"].size == 1:
+        data_copy = lccs_id.copy()
+        data_copy["time"] = lccs_id["time"] + np.timedelta64(1, "D")
+        lccs_id = xr.concat((lccs_id, data_copy), dim="time")
+
+    lccs_id = lccs_id.interp(
+        time=pd.date_range(time_range[0], time_range[1], freq=timestep),
+        method="nearest",
+        kwargs={"fill_value": "extrapolate", "bounds_error": False},
+    )
+
+    landcover_lookup_table = get_landcover_table(cci_dataset)
+    igbp_lookup_table = get_lccs_to_igbp_table()
+
+    return {
+        "LCCS_landcover": np.array(
+            [landcover_lookup_table[_id] for _id in lccs_id.to_numpy()]
+        ),
+        "IGBP_veg_long": np.array(
+            [igbp_lookup_table[_id] for _id in lccs_id.to_numpy()]
+        ),
+    }
+
```

Comment:
Since `LCCS_landcover` is unused in the code, can this function only returns `IGBP_veg_long` as a np.array()?

Reply:
Well, I thought it would be good to keep it in for now, also based on the discussion in https://github.com/EcoExtreML/STEMMUS_SCOPE/issues/138.
It's easier than having to dive into the code again later, to then add it in. 

It would also be nice (in general) if these land cover data could be in the output netCDF, for provenance sake.

Reply:
Yes, when the converter is implemented, it can be added as attrs there. Also, it should be added to the doc as mentioned in #78.